lnwallet/chainfee/rates.go

```diff
@@ -70,6 +70,12 @@ func (s SatPerKWeight) FeeForWeight(wu int64) btcutil.Amount {
 	return btcutil.Amount(s) * btcutil.Amount(wu) / 1000
 }
 
+// FeeForVByte calculates the fee resulting from this fee rate and the given
+// size in vbytes (vb).
+func (s SatPerKWeight) FeeForVByte(vb int64) btcutil.Amount {
```

Comment:
So we have types for the fee rate units, but not the size/weight units. Perhaps we should consider creating `TxWeight` and `TxVsize` types. Callers can still always just go brrrr and do a type cast, but if `SizeUpperBound` returned `TxVsize`, then there would've been an initial compliation failure for the erroneous call to `FeeForWeight`.

Reply:
Definitely want to second the idea of using newtypes (`type Outer Inner` NOT `type Outer = Inner`) as a means of unit management. As a general rule, numbers that can be expressed in multiple units should be typed differently. Unitless numbers should be saved for strict computation. Otherwise they should be given semantic wrappers while they get marshaled from place to place.

Concretely that means that when measuring tx sizes the functions that return the size should return it in a semantic wrapper that specifies the unit. In this case that means:

```go
func (s SatPerKWeight) FeeForVByte(vs TxVsize) Amount { ... }

 // as an aside although this multi-return type is sus 👇🏼
func (t WitnessType) SizeUpperBound() (TxWeight, bool, error) { ... }

// some conversions to include in the API to make it not a PITA
func (wu TxWeight) ToVsize() TxVsize { ... }
func (vb TxVsize) ToWeight() TxWeight { ... }
```

Reply:
> // as an aside, this multi-return type is sus 👇🏼
func (t WitnessType) SizeUpperBound() (TxWeight, bool, error) { ... }

Yeah also don't like this - seems there's some refactoring work to do, but I'll leave it to future PRs - added a TODO to describe the issue, plus we need to fix it for taproot outputs.